src/commands/synthetics/api.ts

```diff
@@ -259,27 +265,43 @@ const createMobileVersion = (request: (args: AxiosRequestConfig) => AxiosPromise
   return resp.data
 }
 
-type RetryPolicy = (retries: number, error: AxiosError) => number | undefined
+const retryWithJitter = (delay: number = DELAY_FOR_TOO_MANY_REQUESTS) => delay + Math.floor(Math.random() * delay)
+
+export type OptionalRetries = {
```

Comment:
**nit** the object is singular
```suggestion
export type RetryPolicy = {
```

Reply:
For this one i want to express in some way that these are optional 🤔 and i feel like it's less of a policy and more configuration for one 🤔

Reply:
You have 2 levels of optionality, and IMO none of them deserves to have "optional" in the name:
- The 1st level is that you can omit the parameter altogether: `determineRetryDelay(retries, error)`
  - For this level, the "optionality" is conveyed by the type, and that's enough.
- The 2nd level is that each property in the object is optional... but it's actually only a boolean (which is optional for simplicity). In the end it's only answering "do you want to retry or not?".

Also, IMO there is a naming conflict between `retries: number` and `optionalRetries: OptionalRetries`. Why would adding an "optional" prefix make `: number` become `: OptionalRetries`?

---

As you said, you are _configuring_ the behavior for retries given the status code. So this could be named "RetryConfiguration"... but I think we use "configuration" far too much already.

And to come back to my suggestion, your refactor is exactly changing one implementation of retry policy with another implementation of retry policy. See [this](https://docs.servicenow.com/bundle/tokyo-application-development/page/administer/flow-designer/concept/retry-policy.html) and [this](https://www.webhooks.io/docs/relay/retry-policies/).

Reply:
hmm okay those are good points 🤔